gadget/gadget.go

```diff
@@ -962,22 +964,28 @@ func LaidOutSystemVolumeFromGadget(gadgetRoot, kernelRoot string, model Model) (
 
 	// find the volume with the system-boot role on it, we already validated
 	// that the system-* roles are all on the same volume
-	for _, vol := range info.Volumes {
+	for name, vol := range info.Volumes {
+		// layout all volumes saving them
+		lvol, err := LayoutVolume(gadgetRoot, kernelRoot, vol, constraints)
+		if err != nil {
+			return nil, nil, err
+		}
+		all[name] = lvol
+		// check if this volume is the boot volume using the system-boot volume
+		// to identify it
 		for _, structure := range vol.Structure {
-			// use the system-boot role to identify the system volume
 			if structure.Role == SystemBoot {
-				pvol, err := LayoutVolume(gadgetRoot, kernelRoot, vol, constraints)
-				if err != nil {
-					return nil, err
-				}
-
-				return pvol, nil
+				system = lvol
```

Comment:
Should we check that `system == nil` before doing this assignment? I guess we expect to find a single system volume anyway, but is it possible (maybe due to a gadget misconfiguration) that there are more than one?

If that is possible, the old code here would return the *first* one found, while with the new code we'd be returning the *last* one.

Reply:
This actually is impossible, as the gadget.yaml must have passed validation above with ReadInfoAndValidate, but I some paranoia is justified as we have had bad gadget bugs before where folks couldn't refresh/update their gadget, so explicitly failing with an internal error here feels like the right thing to do